.github/workflows/release.yaml

```diff
@@ -0,0 +1,160 @@
+          token: ${{ steps.app-token.outputs.token }}
+      - name: Install semver tool
+        run: |
+          curl --fail -LO https://raw.githubusercontent.com/fsaintjacques/semver-tool/3.4.0/src/semver
+          chmod +x ./semver
+      - name: Compose release tag
+        run: |
+          source VERSION
+
+          SHORT_VERSION="$VERSION_MAJOR.$VERSION_MINOR"
+          BASE_VERSION="$SHORT_VERSION.$VERSION_PATCH"
+          TAG="$BASE_VERSION"
+
+          if [ "${{ inputs.version-type }}" == "alpha" ] || [ "${{ inputs.version-type }}" == "beta" ] ; then
+            TAG="$TAG-${{ inputs.version-type }}"
+            TAG_QUERY="$TAG*"
+          elif [ "${{ inputs.version-type }}" == "GA" ]; then
+            TAG_QUERY="$TAG"
+          else
+            echo "Invalid release type: ${{ inputs.version-type }}"
+            exit 1
+          fi
+
+          LAST_MATCH_TAG=$(git tag -l --sort=-v:refname "$TAG_QUERY" | head -n 1)
+
+          if [ "${{ inputs.version-type }}" == "GA" ] && [ "$LAST_MATCH_TAG" != "" ]; then
+            echo "Release $LAST_MATCH_TAG already exists"
+            exit 1
+          fi
+
+          if [ "${{ inputs.version-type }}" == "alpha" ] || [ "${{ inputs.version-type }}" == "beta" ] ; then
+            SUFFIX_NUMBER="1"
+
+            if [ "$LAST_MATCH_TAG" != "" ]; then
+              SUFFIX_NUMBER=$(./semver get prerel $LAST_MATCH_TAG | cut -d'.' -f2)
+              SUFFIX_NUMBER=$((SUFFIX_NUMBER + 1))
+            fi
+
+            TAG="$TAG.$SUFFIX_NUMBER"
+          fi
+
+          echo "SHORT_VERSION=$SHORT_VERSION" >> $GITHUB_ENV
+          echo "BASE_VERSION=$BASE_VERSION" >> $GITHUB_ENV
+          echo "RELEASE_TAG=$TAG" >> $GITHUB_ENV
+          echo "RELEASE_BRANCH=release/$TAG" >> $GITHUB_ENV
+      - name: Validate ${{ env.RELEASE_TAG }} tag
+        run: ./semver validate ${{ env.RELEASE_TAG }}
+      - name: Check if version ${{ env.BASE_VERSION }} is on the right branch
+        run: |
+          source VERSION
+
+          # Need to fetch dev branches first
+          git fetch --no-tags origin HEAD '+refs/heads/development/*:refs/remotes/origin/development/*'
+
+          GIT_REMOTE=$(git remote)
+          BASE_BRANCH=$(git branch -a --list ${GIT_REMOTE}/development/${VERSION_MAJOR}.${VERSION_MINOR})
+          if ! git merge-base --is-ancestor HEAD ${BASE_BRANCH}; then
+            echo "Version commit is not included in base branch ${BASE_BRANCH}"
+            exit 1
+          fi
+      - name: Prepare branch
+        run: git checkout -b ${{ env.RELEASE_BRANCH }}
+      - name: Set the new version
+        run: |
+          if [ -z "${NEW_VERSION_SUFFIX}" ]; then
+            NEW_VERSION_SUFFIX=$(./semver get prerel ${{ env.RELEASE_TAG }})
+            [ -n "$NEW_VERSION_SUFFIX" ] && NEW_VERSION_SUFFIX="-$NEW_VERSION_SUFFIX"
+          fi
+          sed -i "s/VERSION_SUFFIX=.*/VERSION_SUFFIX=$NEW_VERSION_SUFFIX/" VERSION
+      - name: Commit with requested version and push
```

Comment:
Oh btw for MetalK8s for GA we also have to update changelog file

Reply:
You have to handle it before the Release commit as well to remove the `(in development)` suffix

Reply:
Hey!!! I still not committed the changes ;)

Reply:
Done!